src/main/java/Task.java

```diff
@@ -0,0 +1,51 @@
+public class Task {
```

Comment:
The task creation logic could be simplified further by encapsulating task type creation into separate factory methods, reducing the size of the addTask method and improving code readability.